### PR TITLE
Despawn linked traps on use

### PR DESCRIPTION
**Pull request by GuybrushGit**
https://github.com/LightsHope/issues/issues/161

Traps linked to loot objects were not consuming charges when triggered and never despawned, this happens for example on Blood of Heroes: the elites spawn every time someone loots it instead of only on the first loot.

Since they now despawn, adjusted the respawn timer of some linked traps to be lower than the object that triggers them.

### CODE REVIEW

**Comment by bladezLH** [comment]
Why not have the gameobject spawn the linked trap on respawn rather than depending on two independent timers?

**Comment by GuybrushGit** [comment]
Something like this?

File: src/game/Objects/GameObject.cpp [hunk 1]
```diff
@@ -1161,10 +1166,38 @@ void GameObject::TriggerLinkedGameObject(Unit* target)
...
+    GameObjectInfo const* trapInfo = sGOStorage.LookupEntry<GameObjectInfo>(trapEntry);
+    if (!trapInfo || trapInfo->type != GAMEOBJECT_TYPE_TRAP)
+        return;
+
+    float range = 0.5f;
+
+    // search nearest linked GO
+    GameObject* trapGO = NULL;
+    {
+        // search closest with base of used GO, using max range of trap spell as search radius
```
Comment (bladezLH) [inline src/game/Objects/GameObject.cpp:1189]: max range is 0.5f?

Comment (GuybrushGit) [reply src/game/Objects/GameObject.cpp:1189]: Its the same range used when triggering it:
https://github.com/GuybrushGit/core/blob/f1330ef6c625d4d05ae63140f49f8c29dfef4be0/src/game/Objects/GameObject.cpp#L1149-L1152

**Pull request merged**